apps/web/src/app/api/flags/route.ts

```diff
@@ -0,0 +1,7 @@
+import { NextResponse } from 'next/server';
+import { getAllFlags } from '~/utils/feature-flags';
+
+export async function GET() {
```

Comment:
should fetching of the feature flags maybe be done in the middleware layer?

Reply:
It could be done there but also really only tryna keep you from accessing anything other than `/` and `/waitlist`.

So we could move all the logic to the middleware for evaluation but I'll have to play around with that.

Reply:
Fair enough.